src/actions/bookmarks.js

```diff
@@ -54,91 +35,79 @@ export function createBookmark({ slug, host, is_external, title, desc, thumbnail
   return (dispatch) => {
     const { token, userId } = getNecessaryInfo()
     const axiosInstance = getAxiosInstance(token)
-    return new Promise((resolve, reject) => {
-      const path = `${apiEndpoints.users}/${userId}/${apiEndpoints.bookmarks}`
-      const url = formAPIURL(path)
-      axiosInstance.post(url,
-        {
-          slug,
-          host,
-          is_external,
-          title,
-          desc,
-          thumbnail,
-          category,
-          published_date: DateStringToTimeStamp(published_date),
-          authors,
-        })
-        .then((res) => {
-          const { id } = _.get(res, 'data.record', '')
-          resolve(id)
-        })
-        .catch((error) => {
-          const errorInfo = extractErrorInfo(error)
-          dispatch(errorHandler(errorInfo))
-          reject(errorInfo)
-        })
-    })
+    const path = `${apiEndpoints.users}/${userId}/${apiEndpoints.bookmarks}`
+    const url = formAPIURL(path)
+    return axiosInstance.post(url,
+      {
+        slug,
+        host,
+        is_external,
+        title,
+        desc,
+        thumbnail,
+        category,
+        published_date: DateStringToTimeStamp(published_date),
+        authors,
+      })
+      .then((res) => {
+        const { id } = _.get(res, 'data.record', '')
+        return id
+      })
+      .catch((error) => {
+        dispatch(errorHandler(error))
+        throw error
+      })
   }
 }
 
 export function getBookmarks(offset, limit, sort) {
   return (dispatch) => {
     const { token, userId } = getNecessaryInfo()
     const axiosInstance = getAxiosInstance(token)
-    return new Promise((resolve, reject) => {
-      const path = `${apiEndpoints.users}/${userId}/${apiEndpoints.bookmarks}?offset=${offset}&limit=${limit}&sort=${sort}`
-      const url = formAPIURL(path)
-      axiosInstance.get(url)
-        .then((res) => {
-          const data = _.get(res, 'data.records', '')
-          const total = _.get(res, 'data.meta.total', '')
-          resolve({ data, total })
-        })
-        .catch((error) => {
-          const errorInfo = extractErrorInfo(error)
-          dispatch(errorHandler(errorInfo))
-          reject()
-        })
-    })
+    const path = `${apiEndpoints.users}/${userId}/${apiEndpoints.bookmarks}?offset=${offset}&limit=${limit}&sort=${sort}`
+    const url = formAPIURL(path)
+    return axiosInstance.get(url)
+      .then((res) => {
+        const data = _.get(res, 'data.records', '')
+        const total = _.get(res, 'data.meta.total', '')
+        return { data, total }
```

Comment:
Why do not you store `data` and  `total` in the redux state?
Do you want to fetch bookmark list every time when the user is on the bookmark list page?

Reply:
Fixed it. Now, users will fetch data from redux state first.